x/gov/context.go

```diff
@@ -46,3 +52,13 @@ func (a Authenticate) HasAddress(ctx weave.Context, addr weave.Address) bool {
 	}
 	return false
 }
+
+func withProposal(ctx weave.Context, proposal *Proposal, proposalID []byte) weave.Context {
+	return context.WithValue(ctx, contextKeyProposal, proposalWrapper{proposal: proposal, proposalID: proposalID})
+}
+
+// CtxProposal reads the context from the proposal (only set for executor)
+func CtxProposal(ctx weave.Context) (*Proposal, []byte) {
+	val, _ := ctx.Value(contextKeyProposal).(proposalWrapper)
+	return val.proposal, val.proposalID
```

Comment:
This will panic if not set, the previous version set nil.

Please:
1. check if val == nil and return nil, nil explicitly
2. explain the meaning of the []byte return value in the godoc of the function

Reply:
It does not panic

Reply:
Sorry, I missed the cast... `.(proposalWrapper)`, not a pointer.
You are right, my bad.